### ISSUE TITLE
Missing "_ida_netnode.netnode_exist" function in IDA Pro 7.5.201028.

### ISSUE THREAD

**Issue by c0d3z3r0** [arizvisa/ida-minsc#137]
`netnode_exist` does not seem to exist in v7.5.201028.

~~~
CRITICAL:root:internal : Error trying to import module `netnode` from C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_netnode.py.
Traceback (most recent call last):
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\idapythonrc.py", line 148, in load_module
    res = self.new_api(name, path)
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\idapythonrc.py", line 125, in new_api
    res = super(cls, self).new_api(fullname, path)
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\idapythonrc.py", line 67, in new_api
    return self.imp.load_module(modulename, file, path, description)
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_netnode.py", line 17, in <module>
    class netnode(object):
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\_netnode.py", line 42, in netnode
    exist, exist_name = _ida_netnode.exist, _ida_netnode.netnode_exist
AttributeError: 'module' object has no attribute 'netnode_exist'
~~~

Edit:

Interstingly various files match that pattern... wtf.

~~~
IDA SDK and Tools/idasdk75/module/java/npool.cpp:  if ( can_create && !netnode_exist(n) )
IDA SDK and Tools/idasdk75/module/java/npool.cpp:    QASSERT(10333, netnode_exist(n));
IDA SDK and Tools/idasdk75/include/netnode.hpp:idaman bool ida_export netnode_exist(const netnode &n);
IDA SDK and Tools/idasdk75/include/netnode.hpp:  friend bool exist(const netnode &n) { return netnode_exist(n); }
...
grep: python/2/ida_32/_ida_nalt.pyd: binary file matches
grep: python/2/ida_32/_ida_netnode.pyd: binary file matches
grep: python/2/ida_64/_ida_nalt.pyd: binary file matches
grep: python/2/ida_64/_ida_netnode.pyd: binary file matches
...
grep: ida64.dll: binary file matches
grep: ida64.exe: binary file matches
grep: idat64.exe: binary file matches
...
~~~

**Comment by arizvisa** [comment]
Ah. Weird. I thought I fixed that in commit ff3b1a907f4e05934cadfaece64a2c2c0ad65b70. Give me a moment. I'm on it.

**Comment by arizvisa** [comment]
As a temporary fix, you can modify base/_netnode.py.

You'll see this code around line number 42. 
```
    # these apis exist in 5.6 of the sdk, but there's
    # definitely a chance that they're not in IDAPython.
    if idaapi.__version__ > 7.1:
        exist, exist_name = _ida_netnode.exist, _ida_netnode.netnode_exist
    else:
        exist, exist_name = _ida_netnode.exist, internal.utils.fcompose(internal.utils.frpartial(_ida_netnode.new_netnode, False, 0), _ida_netnode.netnode_index, internal.utils.fpartial(operator.ne, idaapi.BADADDR))
```

Just change it to something like the following which forces the second conditional to be used.
```
    # these apis exist in 5.6 of the sdk, but there's
    # definitely a chance that they're not in IDAPython.
    if False:
        exist, exist_name = _ida_netnode.exist, _ida_netnode.netnode_exist
    else:
        exist, exist_name = _ida_netnode.exist, internal.utils.fcompose(internal.utils.frpartial(_ida_netnode.new_netnode, False, 0), _ida_netnode.netnode_index, internal.utils.fpartial(operator.ne, idaapi.BADADDR))
```

Are you using the Python2 interpreter in IDA, or the default Python3? Also, as I'm currently using IDA 7.6.210319, can you give me the output of:
```
import _ida_netnode
print(dir(_ida_netnode))
```

I'm probably going to fix this by explicitly testing to see if `_ida_netnode.netnode_exist` exists, and falling back to the second conditional if it doesn't...

**Comment by arizvisa** [comment]
Oh, also the output the following since you found a reference to it in "ida_nalt" too.
```
import _ida_nalt
print(dir(_ida_nalt))
```

**Comment by c0d3z3r0** [comment]
The temporary fix seems to do the job.

~~~
Python>import _ida_netnode
print(dir(_ida_netnode))
['BADNODE', 'SWIG_PYTHON_LEGACY_BOOL', 'SWIG_PyInstanceMethod_New', '__doc__', '__file__', '__name__', '__package__', 'cvar', 'delete_netnode', 'exist', 'netnode___eq__', 'netnode___ne__', 'netnode_altdel', 'netnode_altdel_all', 'netnode_altdel_ea', 'netnode_altdel_idx8', 'netnode_altfirst', 'netnode_altfirst_idx8', 'netnode_altlast', 'netnode_altlast_idx8', 'netnode_altnext', 'netnode_altnext_idx8', 'netnode_altprev', 'netnode_altprev_idx8', 'netnode_altset', 'netnode_altset_ea', 'netnode_altset_idx8', 'netnode_altshift', 'netnode_altval', 'netnode_altval_ea', 'netnode_altval_idx8', 'netnode_blobsize', 'netnode_blobsize_ea', 'netnode_chardel', 'netnode_chardel_ea', 'netnode_chardel_idx8', 'netnode_charfirst', 'netnode_charfirst_idx8', 'netnode_charlast', 'netnode_charlast_idx8', 'netnode_charnext', 'netnode_charnext_idx8', 'netnode_charprev', 'netnode_charprev_idx8', 'netnode_charset', 'netnode_charset_ea', 'netnode_charset_idx8', 'netnode_charshift', 'netnode_charval', 'netnode_charval_ea', 'netnode_charval_idx8', 'netnode_copyto', 'netnode_create', 'netnode_delblob', 'netnode_delblob_ea', 'netnode_delvalue', 'netnode_eadel', 'netnode_eadel_idx8', 'netnode_eaget', 'netnode_eaget_idx', 'netnode_eaget_idx8', 'netnode_easet', 'netnode_easet_idx', 'netnode_easet_idx8', 'netnode_end', 'netnode_get_name', 'netnode_getblob', 'netnode_getblob_ea', 'netnode_getclob', 'netnode_hashdel', 'netnode_hashdel_all', 'netnode_hashfirst', 'netnode_hashlast', 'netnode_hashnext', 'netnode_hashprev', 'netnode_hashset', 'netnode_hashset_buf', 'netnode_hashset_idx', 'netnode_hashstr', 'netnode_hashstr_buf', 'netnode_hashval', 'netnode_hashval_long', 'netnode_index', 'netnode_kill', 'netnode_long_value', 'netnode_lower_bound', 'netnode_lower_bound_ea', 'netnode_lower_bound_idx8', 'netnode_moveto', 'netnode_next', 'netnode_prev', 'netnode_rename', 'netnode_set', 'netnode_set_long', 'netnode_setblob', 'netnode_setblob_ea', 'netnode_start', 'netnode_supdel', 'netnode_supdel_all', 'netnode_supdel_ea', 'netnode_supdel_idx8', 'netnode_supdel_range', 'netnode_supdel_range_idx8', 'netnode_supfirst', 'netnode_supfirst_idx8', 'netnode_suplast', 'netnode_suplast_idx8', 'netnode_supnext', 'netnode_supnext_idx8', 'netnode_supprev', 'netnode_supprev_idx8', 'netnode_supset', 'netnode_supset_ea', 'netnode_supset_idx8', 'netnode_supshift', 'netnode_supstr', 'netnode_supstr_ea', 'netnode_supstr_idx8', 'netnode_supval', 'netnode_supval_ea', 'netnode_supval_idx8', 'netnode_swiginit', 'netnode_swigregister', 'netnode_valobj', 'netnode_valstr', 'netnode_value_exists', 'new_netnode']
Python>import _ida_nalt
print(dir(_ida_nalt))
['AFL_ALIGNFLOW', 'AFL_BNOT0', 'AFL_BNOT1', 'AFL_COLORED', 'AFL_FIXEDSPD', 'AFL_HIDDEN', 'AFL_LIB', 'AFL_LINNUM', 'AFL_LNAME', 'AFL_LZERO0', 'AFL_LZERO1', 'AFL_MANUAL', 'AFL_NOBRD', 'AFL_NORET', 'AFL_NOTCODE', 'AFL_NOTPROC', 'AFL_PUBNAM', 'AFL_RETFP', 'AFL_SIGN0', 'AFL_SIGN1', 'AFL_TERSESTR', 'AFL_TI', 'AFL_TI0', 'AFL_TI1', 'AFL_TILCMT', 'AFL_USEMODSP', 'AFL_USERSP', 'AFL_USERTI', 'AFL_WEAKNAM', 'AFL_ZSTROFF', 'AP_ALLOWDUPS', 'AP_ARRAY', 'AP_IDXBASEMASK', 'AP_IDXBIN', 'AP_IDXDEC', 'AP_IDXHEX', 'AP_IDXOCT', 'AP_INDEX', 'AP_SIGNED', 'BPU_1B', 'BPU_2B', 'BPU_4B', 'MAXSTRUCPATH', 'NALT_ABSBASE', 'NALT_AFLAGS', 'NALT_ALIGN', 'NALT_COLOR', 'NALT_CREF_FROM', 'NALT_CREF_TO', 'NALT_DREF_FROM', 'NALT_DREF_TO', 'NALT_ENUM0', 'NALT_ENUM1', 'NALT_GR_LAYX', 'NALT_LINNUM', 'NALT_PURGE', 'NALT_STRTYPE', 'NALT_STRUCT', 'NALT_SWITCH', 'NSUP_ARGEAS', 'NSUP_ARRAY', 'NSUP_CMT', 'NSUP_CUSTDT', 'NSUP_FOP1', 'NSUP_FOP2', 'NSUP_FOP3', 'NSUP_FOP4', 'NSUP_FOP5', 'NSUP_FOP6', 'NSUP_FOP7', 'NSUP_FOP8', 'NSUP_FTAILS', 'NSUP_GROUP', 'NSUP_GROUPS', 'NSUP_GR_INFO', 'NSUP_GR_LAYT', 'NSUP_JINFO', 'NSUP_LLABEL', 'NSUP_MANUAL', 'NSUP_OMFGRP', 'NSUP_OPTYPES', 'NSUP_OREF0', 'NSUP_OREF1', 'NSUP_OREF2', 'NSUP_OREF3', 'NSUP_OREF4', 'NSUP_OREF5', 'NSUP_OREF6', 'NSUP_OREF7', 'NSUP_ORIGFMD', 'NSUP_POINTS', 'NSUP_REF0', 'NSUP_REF1', 'NSUP_REF2', 'NSUP_REF3', 'NSUP_REF4', 'NSUP_REF5', 'NSUP_REF6', 'NSUP_REF7', 'NSUP_REGARG', 'NSUP_REGVAR', 'NSUP_REPCMT', 'NSUP_SEGTRANS', 'NSUP_STROFF0', 'NSUP_STROFF1', 'NSUP_SWITCH', 'NSUP_TYPEINFO', 'NSUP_XREFPOS', 'PATCH_TAG', 'POF_VALID_AFLAGS', 'POF_VALID_TI', 'REFINFO_CUSTOM', 'REFINFO_NOBASE', 'REFINFO_PASTEND', 'REFINFO_RVAOFF', 'REFINFO_SIGNEDOP', 'REFINFO_SUBTRACT', 'REFINFO_TYPE', 'RIDX_ABINAME', 'RIDX_ARCHIVE_PATH', 'RIDX_C_MACROS', 'RIDX_DBG_BINPATHS', 'RIDX_DUALOP_GRAPH', 'RIDX_DUALOP_TEXT', 'RIDX_FILE_FORMAT_NAME', 'RIDX_GROUPS', 'RIDX_H_PATH', 'RIDX_IDA_VERSION', 'RIDX_INCLUDE', 'RIDX_MD5', 'RIDX_NOTEPAD', 'RIDX_PROBLEMS', 'RIDX_SELECTED_EXTLANG', 'RIDX_SELECTORS', 'RIDX_SHA256', 'RIDX_SMALL_IDC', 'RIDX_SMALL_IDC_OLD', 'RIDX_SRCDBG_PATHS', 'RIDX_SRCDBG_UNDESIRED', 'RIDX_STR_ENCODINGS', 'STRENC_DEFAULT', 'STRENC_NONE', 'STRLYT_MASK', 'STRLYT_PASCAL1', 'STRLYT_PASCAL2', 'STRLYT_PASCAL4', 'STRLYT_SHIFT', 'STRLYT_TERMCHR', 'STRTYPE_C', 'STRTYPE_C_16', 'STRTYPE_C_32', 'STRTYPE_LEN2', 'STRTYPE_LEN2_16', 'STRTYPE_LEN4', 'STRTYPE_LEN4_16', 'STRTYPE_PASCAL', 'STRTYPE_PASCAL_16', 'STRTYPE_TERMCHR', 'STRWIDTH_1B', 'STRWIDTH_2B', 'STRWIDTH_4B', 'STRWIDTH_MASK', 'SWIG_PYTHON_LEGACY_BOOL', 'SWIG_PyInstanceMethod_New', 'SWI_CUSTOM', 'SWI_DEFRET', 'SWI_DEF_IN_TBL', 'SWI_ELBASE', 'SWI_HXNOLOWCASE', 'SWI_INDIRECT', 'SWI_J32', 'SWI_JMPINSN', 'SWI_JMP_INV', 'SWI_JSIZE', 'SWI_RESERVED', 'SWI_SELFREL', 'SWI_SEPARATE', 'SWI_SHIFT_MASK', 'SWI_SIGNED', 'SWI_SPARSE', 'SWI_STDTBL', 'SWI_SUBTRACT', 'SWI_V32', 'SWI_VERSION', 'SWI_VSIZE', 'SWI_VSPLIT', '__doc__', '__file__', '__name__', '__package__', 'add_encoding', 'array_parameters_t_alignment_get', 'array_parameters_t_alignment_set', 'array_parameters_t_flags_get', 'array_parameters_t_flags_set', 'array_parameters_t_lineitems_get', 'array_parameters_t_lineitems_set', 'array_parameters_t_swiginit', 'array_parameters_t_swigregister', 'clr__bnot0', 'clr__bnot1', 'clr__invsign0', 'clr__invsign1', 'clr_abits', 'clr_align_flow', 'clr_colored_item', 'clr_fixed_spd', 'clr_has_lname', 'clr_has_ti', 'clr_has_ti0', 'clr_has_ti1', 'clr_libitem', 'clr_lzero0', 'clr_lzero1', 'clr_noret', 'clr_notcode', 'clr_notproc', 'clr_retfp', 'clr_terse_struc', 'clr_tilcmt', 'clr_usemodsp', 'clr_usersp', 'clr_userti', 'clr_zstroff', 'custom_data_type_ids_fids_array___getitem__', 'custom_data_type_ids_fids_array___len__', 'custom_data_type_ids_fids_array___setitem__', 'custom_data_type_ids_fids_array_data_get', 'custom_data_type_ids_fids_array_swiginit', 'custom_data_type_ids_fids_array_swigregister', 'custom_data_type_ids_t___getFids', 'custom_data_type_ids_t_dtid_get', 'custom_data_type_ids_t_dtid_set', 'custom_data_type_ids_t_fids_get', 'custom_data_type_ids_t_fids_set', 'custom_data_type_ids_t_set', 'custom_data_type_ids_t_swiginit', 'custom_data_type_ids_t_swigregister', 'cvar', 'dbg_get_input_path', 'del_absbase', 'del_aflags', 'del_alignment', 'del_array_parameters', 'del_custom_data_type_ids', 'del_encoding', 'del_ind_purged', 'del_item_color', 'del_op_tinfo', 'del_refinfo', 'del_source_linnum', 'del_str_type', 'del_switch_info', 'del_switch_parent', 'del_tinfo', 'delete_array_parameters_t', 'delete_custom_data_type_ids_fids_array', 'delete_custom_data_type_ids_t', 'delete_enum_const_t', 'delete_imports', 'delete_opinfo_t', 'delete_printop_t', 'delete_refinfo_t', 'delete_strpath_ids_array', 'delete_strpath_t', 'delete_switch_info_t', 'ea2node', 'encoding_from_strtype', 'enum_const_t_serial_get', 'enum_const_t_serial_set', 'enum_const_t_swiginit', 'enum_const_t_swigregister', 'enum_const_t_tid_get', 'enum_const_t_tid_set', 'enum_import_names', 'find_custom_refinfo', 'get_absbase', 'get_aflags', 'get_alignment', 'get_archive_path', 'get_array_parameters', 'get_asm_inc_file', 'get_custom_data_type_ids', 'get_custom_refinfo', 'get_default_encoding_idx', 'get_encoding_bpu', 'get_encoding_name', 'get_encoding_qty', 'get_gotea', 'get_ids_modnode', 'get_imagebase', 'get_import_module_name', 'get_import_module_qty', 'get_ind_purged', 'get_input_file_path', 'get_item_color', 'get_op_tinfo', 'get_outfile_encoding_idx', 'get_refinfo', 'get_reftype_by_size', 'get_root_filename', 'get_source_linnum', 'get_str_encoding_idx', 'get_str_term1', 'get_str_term2', 'get_str_type', 'get_str_type_code', 'get_str_type_prefix_length', 'get_strtype_bpu', 'get_switch_info', 'get_switch_parent', 'get_tinfo', 'getnode', 'has_lname', 'has_ti', 'has_ti0', 'has_ti1', 'hide_border', 'hide_item', 'is__bnot0', 'is__bnot1', 'is__invsign0', 'is__invsign1', 'is_align_flow', 'is_colored_item', 'is_finally_visible_item', 'is_fixed_spd', 'is_hidden_border', 'is_hidden_item', 'is_libitem', 'is_lzero0', 'is_lzero1', 'is_noret', 'is_notcode', 'is_notproc', 'is_pascal', 'is_reftype_target_optional', 'is_retfp', 'is_terse_struc', 'is_tilcmt', 'is_usersp', 'is_userti', 'is_visible_item', 'is_zstroff', 'new_array_parameters_t', 'new_custom_data_type_ids_fids_array', 'new_custom_data_type_ids_t', 'new_enum_const_t', 'new_opinfo_t', 'new_printop_t', 'new_refinfo_t', 'new_strpath_ids_array', 'new_strpath_t', 'new_switch_info_t', 'node2ea', 'opinfo_t_cd_get', 'opinfo_t_cd_set', 'opinfo_t_ec_get', 'opinfo_t_ec_set', 'opinfo_t_path_get', 'opinfo_t_path_set', 'opinfo_t_ri_get', 'opinfo_t_ri_set', 'opinfo_t_strtype_get', 'opinfo_t_strtype_set', 'opinfo_t_swiginit', 'opinfo_t_swigregister', 'opinfo_t_tid_get', 'opinfo_t_tid_set', 'printop_t_aflags_get', 'printop_t_aflags_set', 'printop_t_features_get', 'printop_t_features_set', 'printop_t_flags_get', 'printop_t_flags_set', 'printop_t_get_ti', 'printop_t_is_aflags_initialized', 'printop_t_is_ti_initialized', 'printop_t_set_aflags_initialized', 'printop_t_set_ti_initialized', 'printop_t_suspop_get', 'printop_t_suspop_set', 'printop_t_swiginit', 'printop_t_swigregister', 'printop_t_ti_get', 'printop_t_ti_set', 'refinfo_t_base_get', 'refinfo_t_base_set', 'refinfo_t_flags_get', 'refinfo_t_flags_set', 'refinfo_t_init', 'refinfo_t_is_custom', 'refinfo_t_is_pastend', 'refinfo_t_is_rvaoff', 'refinfo_t_is_signed', 'refinfo_t_is_subtract', 'refinfo_t_is_target_optional', 'refinfo_t_no_base_xref', 'refinfo_t_set_type', 'refinfo_t_swiginit', 'refinfo_t_swigregister', 'refinfo_t_target_get', 'refinfo_t_target_set', 'refinfo_t_tdelta_get', 'refinfo_t_tdelta_set', 'refinfo_t_type', 'rename_encoding', 'retrieve_input_file_crc32', 'retrieve_input_file_md5', 'retrieve_input_file_sha256', 'retrieve_input_file_size', 'set__bnot0', 'set__bnot1', 'set__invsign0', 'set__invsign1', 'set_abits', 'set_absbase', 'set_aflags', 'set_align_flow', 'set_alignment', 'set_archive_path', 'set_array_parameters', 'set_asm_inc_file', 'set_colored_item', 'set_custom_data_type_ids', 'set_default_encoding_idx', 'set_fixed_spd', 'set_gotea', 'set_has_lname', 'set_has_ti', 'set_has_ti0', 'set_has_ti1', 'set_ids_modnode', 'set_imagebase', 'set_item_color', 'set_libitem', 'set_lzero0', 'set_lzero1', 'set_noret', 'set_notcode', 'set_notproc', 'set_op_tinfo', 'set_outfile_encoding_idx', 'set_refinfo', 'set_refinfo_ex', 'set_retfp', 'set_root_filename', 'set_source_linnum', 'set_str_type', 'set_switch_info', 'set_switch_parent', 'set_terse_struc', 'set_tilcmt', 'set_tinfo', 'set_usemodsp', 'set_usersp', 'set_userti', 'set_visible_item', 'set_zstroff', 'strpath_ids_array___getitem__', 'strpath_ids_array___len__', 'strpath_ids_array___setitem__', 'strpath_ids_array_data_get', 'strpath_ids_array_swiginit', 'strpath_ids_array_swigregister', 'strpath_t___getIds', 'strpath_t_delta_get', 'strpath_t_delta_set', 'strpath_t_ids_get', 'strpath_t_ids_set', 'strpath_t_len_get', 'strpath_t_len_set', 'strpath_t_swiginit', 'strpath_t_swigregister', 'switch_info_t_SWITCH_INFO_VERSION', 'switch_info_t__from_ptrval__', 'switch_info_t__get_values_lowcase', 'switch_info_t__set_values_lowcase', 'switch_info_t_assign', 'switch_info_t_clear', 'switch_info_t_custom_get', 'switch_info_t_custom_set', 'switch_info_t_defjump_get', 'switch_info_t_defjump_set', 'switch_info_t_elbase_get', 'switch_info_t_elbase_set', 'switch_info_t_expr_ea_get', 'switch_info_t_expr_ea_set', 'switch_info_t_flags_get', 'switch_info_t_flags_set', 'switch_info_t_get_jrange_vrange', 'switch_info_t_get_jtable_element_size', 'switch_info_t_get_jtable_size', 'switch_info_t_get_lowcase', 'switch_info_t_get_shift', 'switch_info_t_get_version', 'switch_info_t_get_vtable_element_size', 'switch_info_t_has_default', 'switch_info_t_has_elbase', 'switch_info_t_ind_lowcase_get', 'switch_info_t_ind_lowcase_set', 'switch_info_t_is_custom', 'switch_info_t_is_indirect', 'switch_info_t_is_nolowcase', 'switch_info_t_is_sparse', 'switch_info_t_is_subtract', 'switch_info_t_jcases_get', 'switch_info_t_jcases_set', 'switch_info_t_jumps_get', 'switch_info_t_jumps_set', 'switch_info_t_marks_get', 'switch_info_t_marks_set', 'switch_info_t_ncases_get', 'switch_info_t_ncases_set', 'switch_info_t_regdtype_get', 'switch_info_t_regdtype_set', 'switch_info_t_regnum_get', 'switch_info_t_regnum_set', 'switch_info_t_set_elbase', 'switch_info_t_set_expr', 'switch_info_t_set_jtable_element_size', 'switch_info_t_set_jtable_size', 'switch_info_t_set_shift', 'switch_info_t_set_vtable_element_size', 'switch_info_t_startea_get', 'switch_info_t_startea_set', 'switch_info_t_swiginit', 'switch_info_t_swigregister', 'switch_info_t_use_std_table', 'unhide_border', 'unhide_item', 'uses_modsp', 'validate_idb_names']
~~~

**Comment by c0d3z3r0** [comment]
hm, this seems to have to do with netnode, too, since it uses `has`:

~~~
Python>enum.by_identifier(0xff00020f)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\enumeration.py", line 114, in by_identifier
    raise E.EnumerationNotFoundError(u"{:s}.by_identifier({!s}) : Unable to locate the enumeration with the specified identifier ({:#x}).".format(__name__, eid, eid))
internal.exceptions.EnumerationNotFoundError: enumeration.by_identifier(4278190607) : Unable to locate the enumeration with the specified identifier (0xff00020f).
Python>idaapi.get_enum_name(0xff00020f)
'RAM1_01'
~~~

**Comment by arizvisa** [comment]
> hm, this seems to have to do with netnode, too, since it uses `has`:
> 
> ```
> Python>enum.by_identifier(0xff00020f)
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\enumeration.py", line 114, in by_identifier
>     raise E.EnumerationNotFoundError(u"{:s}.by_identifier({!s}) : Unable to locate the enumeration with the specified identifier ({:#x}).".format(__name__, eid, eid))
> internal.exceptions.EnumerationNotFoundError: enumeration.by_identifier(4278190607) : Unable to locate the enumeration with the specified identifier (0xff00020f).
> Python>idaapi.get_enum_name(0xff00020f)
> 'RAM1_01'
> ```

This is using `enumeration.has` which has a differing implementation and is thus a different issue. I'll create a new one for you, and link it though.

**Comment by arizvisa** [comment]
This issue is remedied in PR #138 using the explicit check of the symbol that I mentioned as a temporary fix instead of checking against the IDA version. It seems that the most reliable cross-version method would really be to just assume that the `_ida_netnode.netnode_exist` is never defined and to always use `_ida_netnode.new_netnode` with its "do_create" parameter set to `False`. However, I don't feel right about removing `_ida_netnode.netnode_exist` so I ended up deciding upon using that temporary fix that was mentioned.

If you have an disagreements about this, let me know. I'll prolly merge this later today after I look at your other bug.

**Comment by arizvisa** [comment]
> > hm, this seems to have to do with netnode, too, since it uses `has`:
> > ```
> > Python>enum.by_identifier(0xff00020f)
> > Traceback (most recent call last):
> >   File "<string>", line 1, in <module>
> >   File "C:\users\c0d3\Application Data\Hex-Rays\IDA Pro\ida_minsc\base\enumeration.py", line 114, in by_identifier
> >     raise E.EnumerationNotFoundError(u"{:s}.by_identifier({!s}) : Unable to locate the enumeration with the specified identifier ({:#x}).".format(__name__, eid, eid))
> > internal.exceptions.EnumerationNotFoundError: enumeration.by_identifier(4278190607) : Unable to locate the enumeration with the specified identifier (0xff00020f).
> > Python>idaapi.get_enum_name(0xff00020f)
> > 'RAM1_01'
> > ```
> 
> This is using `enumeration.has` which has a differing implementation and is thus a different issue. I'll create a new one for you, and link it though.

I created issue #139 for this other issue, if you don't mind I'll continue troubleshooting this one over there.

**Comment by arizvisa** [comment]
Closing this issue as it was fixed by PR #138. If the issue still persists, let me know and I can re-open.

Thanks again for your contribution!
